ramda.js

```diff
@@ -4705,6 +4707,42 @@
     R.gte = op(function _gte(a, b) { return a >= b; });
 
 
+    function createMaxMin(comparator, initialVal) {
+        return function(list) {
+            if (arguments.length === 0) {
+                throw noArgsException();
+            }
+            var idx = -1, winner = initialVal, computed;
+            while (++idx < list.length) {
+                computed = +list[idx];
+                if (comparator(computed, winner)) {
+                    winner = computed;
+                }
+            }
+            return winner;
+        };
+    }
+
+    function createMaxMinWith(comparator) {
+        return function(valueComputer, list) {
+            if (!(list && list.length > 0)) {
+                return;
```

Comment:
Well, I guess that's the distinction I was making.  `max`/`min` in our world are ideally 

    :: [Number] -> Number

But we relax it a bit to allow

    :: [CoerceableToNumber] -> Number

But `maxBy`/`minBy` are

    :: (a -> Number) -> [a] -> a

Your description for `max`/`min` sound more like

    :: [CoerceableToNumber] -> CoerceableToNumber

While we could certainly support that, I prefer a `[Number] -> Number` signature that happens to be slightly relaxed.